src/CrosschainDeployScript.sol

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+
+pragma solidity 0.8.20;
+
+import {Script} from "forge-std/Script.sol";
+import {ICrosschainDeployAdapter} from "./interfaces/CrosschainDeployAdapterInterface.sol";
+
+/**
+ * @title Provides a script to allow users to call the multichain deployment contract defined in `CrossChainDeployAdapter` from chainsafe/hardhat-plugin-multichain-deploy, passing it the contract bytecode and constructor arguments.
+ * @author ChainSafe Systems
+ */
+contract CrosschainDeployScript is Script {
+    // this is the address of the original contract defined in chainsafe/hardhat-plugin-multichain-deploy
+    // this address is the same across all chains
+    address private constant CROSS_CHAIN_DEPLOY_CONTRACT_ADDRESS = 0x85d62AD850B322152BF4ad9147bfBF097DA42217;
+
+    /**
+     * @notice this function takes in the contract string, in the form that
+     * @notice `forge`'s `getCode` takes it, along with some other parameters and passes
+     * @notice it along to the `deploy` function of the `CrossChainDeployAdapter`
+     * @notice contract.
+     * @param contractString Contract name in the form of `ContractFile.sol`, if the name of the contract and the file are the same, or `ContractFile.sol:ContractName` if they are different.
+     * @param gasLimit Contract deploy and init gas.
+     * @param salt Entropy for contract address generation.
+     * @param isUniquePerChain True to have unique addresses on every chain.
+     * @param constructorArgs Bytes to add to the deployBytecode, or empty, one per chain.
+     * @param initDatas Bytes to send to the contract after deployment, or empty, one per chain.
+     * @param destinationDomainIDs Sygma Domain IDs of target chains.
+     *
+     *   Users call this function and pass only the function call string as
+     *   `MyContract.sol:MyContract`. The function call string is then parsed
+     *   and the `callData` and `bytesCode` are extracted from it.
+     *   and the contract is deployed on the other chains.
+     */
+    function deployContractOnOtherChains(
```

Comment:
I don't think this wrapper really improves dev ux for using our tool.
I was thinking more to provide interface like:
```solidity
addDeploymentTarget(string network, bytes constructorArgs, initData) //should store target in a map, network should be converted to domain by us, it should be documented how to encode constructor args and initData
encodeInitData(contract, functionName, encodedArgs)
generateSalt()
deploy(contract)
deploy(contract, salt)
deploy(contract, salt, isUniquePerChain)
getDeploymentFee()
computeAddressForChain()
```

and when someone extends it it can do something like:

```solidity
var contract = "MyContract"
var initData = this.calculateInitData(contract, "init", abi.encodePacked("1", "2"))
this.addDeploymentTarget(contract, "mainnet", "0x", initData)

this.deploy(contract)
```